src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx

```diff
@@ -329,7 +334,16 @@ export const TradeInput = memo(({ isCompact }: TradeInputProps) => {
     selectBuyAmountBeforeFeesCryptoPrecision,
   )
 
+  const handleConnect = useCallback(() => {
+    walletDispatch({ type: WalletActions.SET_WALLET_MODAL, payload: true })
+  }, [walletDispatch])
+
   const onSubmit = useCallback(() => {
+    // No preview happening if wallet isn't connected i.e is using the demo wallet
+    if (!isConnected || isDemoWallet) {
+      return handleConnect()
```

Comment:
non-blocking nitpick, but this can work too
```suggestion
      return walletDispatch({ type: WalletActions.SET_WALLET_MODAL, payload: true })
```

Reply:
Prefer to keep `handleConnect` its own for declarativeness 🙏🏽